best-practices/rspec.md

```diff
@@ -0,0 +1,30 @@
+## Rspec style guide
+
+- First #describe what you are doing
+- Then establish the #context
+- #it only expects one thing (within reason)
+- Prefer explicitness - #it #its #specify #subject are nice shortcuts but
+consider whether they are sacrificing readability
+- Having excessive setup (fixtures, database entities, doubles etc.) can cause false-positives. Thus, set up preconditions as close to the test as possible (usually within the context block) and tear them down afterwards. This is really important when working with time for example.
+- Unit tests are fast and effective to work with when doing TDD. These are isolated tests, typically
+for a single class, and should use stubs and spies via test doubles to replace anything that is
+external to the _unit_ being tested. Testing side-effects that occur elsewhere in the system
+is incorrect. Not being able to unit test something is typically a signal that something may be wrong
+with the design of the unit
```

Comment:
So I suspect that we'll have faith wars if we try and put down a definition for _unit_ - personally, I think they should be as small as possible otherwise you are testing the way different pieces of the system interact. Doesn't really matter if we stay with the philosophy of stubbing everything on the boundary of whatever we are calling a unit. Yeah?